setup.cfg

```diff
@@ -22,3 +22,8 @@ ignore_directives = autoattribute,autoclass,autoexception,autofunction,autometho
 
 [darglint]
 docstring_style = numpy
+
+[mypy]
+ignore_missing_imports = True
+scripts_are_modules = True
+show_error_codes = True
```

Comment:
What is the motivation for each of these options?

Reply:
- `ignore_missing_imports`: Prevents errors where dependencies cause errors due to their missing type.
E.g,` Skipping analyzing 'numpy': found module but no type hints or library stubs  [import]`
- `show_error_codes`: Displays the error code at the end of an error, which makes it easier to look it up or if needed ignore an error in a fine-grain manner, e.g. `import` in the above example
- `scripts_are_modules`: Was taken from the [default arguments of the hook](https://github.com/pre-commit/mirrors-mypy/blob/6fd7e2a39e5a8f7c7b14ddcc5290ef69b0c446db/.pre-commit-hooks.yaml#L6) and [does only have an effect on script files without `*.py` ending](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-scripts-are-modules). So I guess we can drop it

Reply:
Also, we might want to add the config in a separate PR so we don't need to pass all the flags to `mypy` when fixing existing errors before we can use the pre-commit hook.